packages/headless/src/features/relative-date-set/relative-date-set-slice.ts

```diff
@@ -0,0 +1,18 @@
+import {createReducer} from '@reduxjs/toolkit';
+import {registerRelativeDate} from './relative-date-actions';
+import {getRelativeDateSetInitialState} from './relative-date-set-state';
+
+export const relativeDateSetReducer = createReducer(
+  getRelativeDateSetInitialState(),
+  (builder) => {
+    builder.addCase(registerRelativeDate, (state, {payload}) => {
+      const {id, relativeDate, absoluteDate} = payload;
+      if (!state[id]) {
+        state[id] = {};
+        return;
+      }
+
+      state[id][absoluteDate] = relativeDate;
```

Comment:
Could it be possible for two different relative dates to collide/have the same absoluteDate, causing an unexpected resolution? For example, 1 week vs. 7 days, where the ranges are:

```
7days ago - now
2 weeks ago - 1 week ago
```

Depending on the order, `1 week ago` might be resolved as `7 days ago` in the controller. Someone rendering the ranges using the controller state would unexpectedly print: `2 weeks ago - 7 days ago`.

If this is possible, it might be safer to store the full range, so that it is possible to compare the start and end, to ensure we are resolving the relative date the developer originally configured.